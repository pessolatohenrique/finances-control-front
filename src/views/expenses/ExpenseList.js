import React from "react";
import moment from "moment";
import {
  Grid,
  Card,
  CardContent,
  Typography,
  List,
  ListItem,
  ListItemText,
} from "@mui/material";

export function ExpenseList({ budget }) {
  return (
    <Grid container spacing={2}>
      {budget?.Expenses.map((row) => {
        return (
          <Grid item xs={4} key={row?.id}>
            <Card data-testid="card-item">
              <CardContent>
                <Typography variant="h6" component="div" color="text.secondary">
                  {row?.Expense?.name}
                </Typography>
                <List variant="body2">
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        {row?.Expense?.userExpenseCategory?.Category?.name}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">R$ {row?.value}</Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        {moment(row?.transaction_date).format("DD/MM/YYYY")}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                </List>
              </CardContent>
            </Card>
          </Grid>
        );
      })}
    </Grid>
  );
}
