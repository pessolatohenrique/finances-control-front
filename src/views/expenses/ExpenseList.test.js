import moment from "moment";
import { render, screen } from "@testing-library/react";
import { ExpenseList } from "./ExpenseList";

const budget = {
  Expenses: [
    {
      id: 1,
      name: "Internet",
      userExpenseCategory: { Category: { name: "Essencial" } },
      UserExpense: {
        value: 300,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
    {
      id: 2,
      name: "Camisetas",
      userExpenseCategory: { Category: { name: "Livre estou" } },
      UserExpense: {
        value: 90,
        transaction_date: moment().format("YYYY-MM-DD"),
      },
    },
  ],
};

describe("Expense List", () => {
  it("should generate snapshot", () => {
    const { container } = render(<ExpenseList budget={budget} />);
    expect(container).toMatchSnapshot();
  });

  it("should generate table", () => {
    render(<ExpenseList budget={budget} />);
    const rowTable = screen.getAllByTestId("card-item");
    expect(rowTable.length).toBe(2);
  });
});
