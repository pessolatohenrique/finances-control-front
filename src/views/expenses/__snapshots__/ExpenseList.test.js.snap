// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List should generate snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Internet
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  Essencial
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  R$ 
                  300
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  17/05/2022
                </p>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Camisetas
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  Livre estou
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  R$ 
                  90
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  17/05/2022
                </p>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
